src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/FilesystemSupplementedRepositoryImpl.java

```diff
@@ -215,9 +220,14 @@ private Type writeFilesToDisk(Path baseDirectory, Type objectToWrite) {
 
 		Predicate<Field> pathFilter = f -> f.getType().equals(Path.class);
 		// now find any members that are of type Path and shuffle them around:
-		Set<Field> pathFields = Arrays.stream(objectToWrite.getClass().getDeclaredFields()).filter(pathFilter)
+		Set<Field> pathFields = Arrays.stream(objectToWrite.getClass().getSuperclass().getDeclaredFields()).filter(pathFilter)
```

Comment:
Avoid getSuperClass if you can.  That gets us into dodgy places.

Reply:
Updated in 3569c6bffc239051ba0b600141464aff94fcdd6e